cfg/Instructions.h

```diff
@@ -405,6 +409,51 @@ class InstructionPtr final {
         return (this->ptr & SYNTHETIC_FLAG) != 0;
     }
 
+    template <class To> struct TaggedValue {
+    private:
+        To *value;
+        bool isValid;
+
+    public:
+        explicit TaggedValue(To *p, bool valid) noexcept : value(p), isValid(valid) {}
+
+        explicit operator bool() const noexcept {
+            return isValid;
+        }
+
+        To *operator->() const noexcept {
+            return value;
+        }
+
+        operator To *() const noexcept {
+            return value;
+        }
+
+        bool operator==(std::nullptr_t) const noexcept {
+            return value == nullptr;
+        }
+
+        bool operator!=(std::nullptr_t) const noexcept {
+            return value != nullptr;
+        }
+    };
+
+    template <class To> TaggedValue<To> as_instruction() {
+        bool isValid = tagMaybeZero() == uint16_t(InsnToTag<To>::value);
+        if (!isValid) {
+            return TaggedValue<To>(nullptr, isValid);
+        }
+        return TaggedValue<To>(reinterpret_cast<To *>(get()), isValid);
```

Comment:
What do you think about this instead? Using `isValid` for the second argument of both constructor calls gave me pause 😅 
```suggestion
        auto *ptr = isValid ? reinterpret_cast<To *>(get()) : nullptr;
        return TaggedValue<To>(ptr, isValid);
```

Reply:
Works for me!